### PR TITLE
Add Issue Forms

### PR DESCRIPTION
**Pull request by MatteoGheza**
As discussed on Zulip, I added issue forms (issue templates but with defined input types and with better UI for the user) for bug reports and feature requests.
As requested, the only required input is the description.
If you want to try this, check [this example repo](https://github.com/MatteoGheza/mixxx-issue-templates-PoC/issues/new/choose).

### CODE REVIEW

**Comment by ronso0** [comment]
Thanks for starting this!

re OS combobox:
considering Mixxx is built on many linux distros which can't be covered by a combobox, I think the OS/version should be a regular line edit. The question should be "What operating system and version are you running?"

**Comment by Holzhaus** [comment]
> Thanks for starting this!
> 
> re OS combobox:
> considering Mixxx is built on many linux distros which can't be covered by a combobox, I think the OS/version should be a regular line edit. The question should be "What operating system and version are you running?"

A lot of users don't know what an OS is, so maybe we should give an example? 

> What operating system and version are you running (e.g. Windows 11, macOS Big Sur, Ubuntu 22.04, ...)?

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug report!
+  - type: textarea
+    id: description
+    attributes:
+      label: Bug Description
+      description: Also tell us, what did you expect to happen?
```
Comment (daschuer) [inline .github/ISSUE_TEMPLATE/bug.yaml:14]:
```suggestion
      description: Tell us what happens and what should happen instead.

```

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      placeholder: Tell us what you see!
```
Comment (daschuer) [inline .github/ISSUE_TEMPLATE/bug.yaml:15]: This seems to be redundant
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: What version of Mixxx are you running?
+  - type: dropdown
+    id: os
+    attributes:
+      label: OS
+      description: What operating system are you running?
```
Comment (daschuer) [inline .github/ISSUE_TEMPLATE/bug.yaml:27]:
```suggestion
      description: What operating system and distribution are you running?

```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: 🚀 Feature Request
+description: |
+  Describe the requested feature here.
```
Comment (daschuer) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:3]:
```suggestion
  Describe describe your use-case that is not yet supported and a possible solution.
```

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you for bringing that up.

**Comment by Holzhaus** [comment]
Please add `sur` (lowercase) to `.codespellignore` to fix the pre-commit issue:

```
.github/ISSUE_TEMPLATE/bug.yaml:27: Sur ==> Sure, sir
```

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: What version of Mixxx are you running?
+  - type: input
+    id: os
+    attributes:
+      label: OS
+      description: What operating system (and distribution if you're on Linux) are you running?
```
Comment (MatteoGheza) [inline .github/ISSUE_TEMPLATE/bug.yaml:26]: What do you think about this?

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      placeholder: Windows 11, macOS Big Sur, Ubuntu 22.04, ...
```
Comment (MatteoGheza) [inline .github/ISSUE_TEMPLATE/bug.yaml:27]: What do you think about this?

**Comment by ronso0** [comment]
@MatteoGheza Would you mind updating your demo repo? Wuld be nice if we could test this 'live'.

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to suggest a new feature!
+  - type: textarea
+    id: description
+    attributes:
+      label: Feature Description
+      description: What feature would you like to see added to Mixxx?
```
Comment (daschuer) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:14]: I think we should swap the title description with this one.

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+name: 🐛 Bug Report
```
Comment (Holzhaus) [inline .github/ISSUE_TEMPLATE/bug.yaml:1]: This caterpillar is easily confused with a snake. What do you think about using 🪲 (U+1FAB2) instead?

**Review by Holzhaus — COMMENTED** [review]
Thanks. IIRC it's also possible to add links, right? If so we could add a third option in addition to "Bug" and "Feature Request": something like "Question" that is a link to the discourse board.

File: .github/ISSUE_TEMPLATE/bug.yaml (hunk 8 above)
Comment (daschuer) [reply .github/ISSUE_TEMPLATE/bug.yaml:1]: Unfortunately I only see a gray rectangular for the proposed alternative on my mobile.

I had also this idea initially, than I realized that  🐛, U+1F41B is called Bug.

**Comment by MatteoGheza** [comment]
> @MatteoGheza Would you mind updating your demo repo? Wuld be nice if we could test this 'live'.

Done. [https://github.com/MatteoGheza/mixxx-issue-templates-PoC/issues/new/choose](https://github.com/MatteoGheza/mixxx-issue-templates-PoC/issues/new/choose)

**Comment by MatteoGheza** [comment]
> Thanks. IIRC it's also possible to add links, right? If so we could add a third option in addition to "Bug" and "Feature Request": something like "Question" that is a link to the discourse board.

Yes, I think we can do something like this (as an optional field).

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    attributes:
+      label: Bug Description
+      description: Tell us what happens and what should happen instead.
+    validations:
+      required: true
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: What version of Mixxx are you running?
```
Comment (Swiftb0y) [inline .github/ISSUE_TEMPLATE/bug.yaml:21]: IMO a placeholder for the version would also make sense to suggest that we'd like to know the entire `major.minor.patch` version triple, not just `major.minor`.

```suggestion
      description: What version of Mixxx are you running?
      placeholder: 2.3.4, 2.4.0, ...
```

File: .github/ISSUE_TEMPLATE/bug.yaml (hunk 8 above)
Comment (Swiftb0y) [reply .github/ISSUE_TEMPLATE/bug.yaml:1]: :bug: was added in Emoji 1.0 while :bettle: was added in 13.0 (2020)... I still prefer :beetle:

File: .github/ISSUE_TEMPLATE/config.yml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
+contact_links:
+  - name: 🤔 Question
+    url: https://mixxx.org/forums
+    about: Please use our forums for questions
```
Comment (MatteoGheza) [inline .github/ISSUE_TEMPLATE/config.yml:5]: > I don't think thats possible.

It's possible to add links (with title and description) as "contact links". What do you think about this @Holzhaus?
Remember that you can try this at https://github.com/MatteoGheza/mixxx-issue-templates-PoC/issues/new/choose

**Comment by Holzhaus** [comment]
> > Thanks. IIRC it's also possible to add links, right? If so we could add a third option in addition to "Bug" and "Feature Request": something like "Question" that is a link to the discourse board.
> 
> I don't think thats possible. This is the closest thing I came up with that *should* validate.
> ```yaml
> name: 🤔 Question
> description: |
>   Please use [our forums](https://mixxx.org/forums) for questions
> body:
>   # body must contain at least one non-markdown field
>   # https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#body-must-contain-at-least-one-non-markdown-field
>   - type: dropdown
>     attributes:
>       label: |
>         Please use [our forums](https://mixxx.org/forums) for questions
>       options:
>         - Please use [our forums](https://mixxx.org/forums) for questions
>     validations:
>       required: true
>   
> ```

Have a look at:
https://github.com/beetbox/beets/blob/master/.github/ISSUE_TEMPLATE/config.yml

And the result:
https://github.com/beetbox/beets/issues/new/choose

File: .github/ISSUE_TEMPLATE/config.yml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
```
Comment (MatteoGheza) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Would you prefer to enable the possibility to add issues without using a template?

File: .github/ISSUE_TEMPLATE/config.yml (hunk 10 above)
Comment (Holzhaus) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Great, exactly what I had in mind! Thanks.

**Comment by Swiftb0y** [comment]
Good catch. I didn't know there was an `ISSUE_TEMPLATE/config.yml`. I was only digging through the documentation of the forms yaml scheme. 
The solution in the newest commit is much better than my proposal.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 10 above)
Comment (Swiftb0y) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Good catch. I didn't know there was an `ISSUE_TEMPLATE/config.yml`. I was only digging through the documentation of the forms yaml scheme. 
This solution is much better than my proposal.

Comment (ronso0) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Nitpick: instead of arbitrary `Open` let's use `Go to Mixxx user forum`?

Comment (ronso0) [reply .github/ISSUE_TEMPLATE/config.yml:5]: if possible at all...

File: .github/ISSUE_TEMPLATE/config.yml (hunk 11 above)
Comment (Holzhaus) [reply .github/ISSUE_TEMPLATE/config.yml:1]: I'm torn on this.

For bugs and feature requests, this is not neccessary. We don't have some complex JIRA-like form with dozens of fields, just a description + affected version + OS.

However, I think we may sometimes want to create issues as TODO items (e.g. for project boards) and then these two options may not be a good fit. Maybe we should add a new type "Task" for those kind of issues? Or could we restrict blank issues for contributors only?

Comment (ronso0) [reply .github/ISSUE_TEMPLATE/config.yml:1]: :+1: "TODO" for contributors only

Comment (Swiftb0y) [reply .github/ISSUE_TEMPLATE/config.yml:1]: > Or could we restrict blank issues for contributors only?

If thats possible, that would be ideal IMO.

If not, we should have blank issues enabled for all IMO.

Comment (MatteoGheza) [reply .github/ISSUE_TEMPLATE/config.yml:1]: I think that with config version we can't do something like that. I'll enable blank issues in the config.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 10 above)
Comment (MatteoGheza) [reply .github/ISSUE_TEMPLATE/config.yml:5]: I think this is not possible... The config file has not more options than this.

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Review by Swiftb0y — CHANGES_REQUESTED** [review]
Did we come to a conclusion on :bug: vs :beetle: @Holzhaus ?

**Review by Holzhaus — APPROVED** [review]
Thank you, this is definitely an nice improvement.

I'd tlike to merge this now - if someone wants to fine-tune the details later on, we can still do this in a follow-up PR. Just waiting for approval by @Swiftb0y.

**Review by Swiftb0y — APPROVED** [review]

**Comment by ronso0** [comment]
Thanks @MatteoGheza
I'm sure this will save time reviewing bug reports

**Pull request merged**